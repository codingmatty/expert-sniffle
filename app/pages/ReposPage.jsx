import React from 'react';

import ReposList from 'components/ReposList';

import './ReposPage.scss'

export default function ReposPage() {
  return (
    <div className="repos-page flex direction-row mx-4 medium:mx-auto">
      <ReposList />
    </div>
  );
}







