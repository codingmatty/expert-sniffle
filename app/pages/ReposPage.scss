.repos-page {
  @media (min-width: 768px) {
    max-width: 738px;
  }

  @media (min-width: 1030px) {
    max-width: 1000px;
  }
}
