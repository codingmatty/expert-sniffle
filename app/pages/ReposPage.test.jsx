import '@testing-library/jest-dom';
import configureStore from 'redux-mock-store';
import React from 'react';
import { render, screen } from '@testing-library/react';
import ReposPage from './ReposPage';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

const mockStore = configureStore([thunk]);

describe('ReposPage', () => {
  describe('loading', () => {
    test('shows loading message', () => {
      render(
        <Provider store={mockStore({ repos: { loading: true } })}>
          <ReposPage />
        </Provider>
      );
      expect(screen.getByText('Loading...')).toBeDefined();
    });
  });

  describe('error', () => {
    test('shows error message', () => {
      const { asFragment } = render(
        <Provider
          store={mockStore({ repos: { error: 'An error occurred', data: [] } })}
        >
          <ReposPage />
        </Provider>
      );
      expect(screen.getByText('An error occurred')).toBeDefined();
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('with data', () => {
    test('empty array', () => {
      const { asFragment } = render(
        <Provider store={mockStore({ repos: { data: [] } })}>
          <ReposPage />
        </Provider>
      );
      expect(
        screen.getByText('No Repos Found for provided token')
      ).toBeDefined();
      expect(asFragment()).toMatchSnapshot();
    });

    test('with list of repos', () => {
      const data = [
        {
          id: 1,
          name: 'Name',
          full_name: 'full/name',
          description: 'Lorem Ipsum Dolor Sit Amet',
          stargazers_count: 1,
          open_issues_count: 3,
          html_url: 'https://github.com/full/name',
        },
        {
          id: 2,
          private: true,
          name: 'Name',
          full_name: 'full/name',
          description: 'Lorem Ipsum Dolor Sit Amet',
          stargazers_count: 1,
          open_issues_count: 3,
          html_url: 'https://github.com/full/name',
        },
        {
          id: 3,
          name: 'Name',
          full_name: 'full/name',
          description: 'Lorem Ipsum Dolor Sit Amet',
          stargazers_count: 1,
          open_issues_count: 3,
          html_url: 'https://github.com/full/name',
        },
      ];
      const { asFragment } = render(
        <Provider store={mockStore({ repos: { data } })}>
          <ReposPage />
        </Provider>
      );
      expect(screen.getAllByTestId('repo-details')).toHaveLength(3);
      expect(screen.getByText('lock')).toBeDefined();
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
