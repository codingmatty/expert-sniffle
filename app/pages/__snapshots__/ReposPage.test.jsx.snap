// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReposPage error shows error message 1`] = `
<DocumentFragment>
  <div
    class="repos-page flex direction-row mx-4 medium:mx-auto"
  >
    <div
      class="repos-list grow-1"
    >
      <div
        class="flex direction-row space-between items-center mb-2"
      >
        <div
          class="flex direction-column"
        >
          <h2
            class="mb-2"
          >
            Repos
          </h2>
          <a
            aria-current="page"
            href="/"
          >
            Update Token
          </a>
        </div>
        <button>
          Reload Repos
        </button>
      </div>
      <ul>
        <p>
          An error occurred
        </p>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ReposPage with data empty array 1`] = `
<DocumentFragment>
  <div
    class="repos-page flex direction-row mx-4 medium:mx-auto"
  >
    <div
      class="repos-list grow-1"
    >
      <div
        class="flex direction-row space-between items-center mb-2"
      >
        <div
          class="flex direction-column"
        >
          <h2
            class="mb-2"
          >
            Repos
          </h2>
          <a
            aria-current="page"
            href="/"
          >
            Update Token
          </a>
        </div>
        <button>
          Reload Repos
        </button>
      </div>
      <ul>
        <p>
          No Repos Found for provided token
        </p>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ReposPage with data with list of repos 1`] = `
<DocumentFragment>
  <div
    class="repos-page flex direction-row mx-4 medium:mx-auto"
  >
    <div
      class="repos-list grow-1"
    >
      <div
        class="flex direction-row space-between items-center mb-2"
      >
        <div
          class="flex direction-column"
        >
          <h2
            class="mb-2"
          >
            Repos
          </h2>
          <a
            aria-current="page"
            href="/"
          >
            Update Token
          </a>
        </div>
        <button>
          Reload Repos
        </button>
      </div>
      <ul>
        <li>
          <div
            class="repo-details pointer space-between mt-4 border border-gray radius-3 p-4 bg-white "
            data-testid="repo-details"
          >
            <div
              class="info"
            >
              <div
                class="text-large"
              >
                <b>
                  Name
                </b>
              </div>
              <div>
                <a
                  href="https://github.com/full/name"
                  target="_blank"
                >
                  full/name 
                  <i
                    class="material-icons "
                  >
                    open_in_new
                  </i>
                </a>
              </div>
            </div>
            <div
              class="star-count flex items-center text-large"
            >
              <i
                class="material-icons mr-2"
              >
                star
              </i>
              1
            </div>
            <div
              class="issue-count flex items-center text-large"
            >
              <i
                class="material-icons mr-2"
              >
                error_outline
              </i>
              3
            </div>
            <p
              class="mt-4 description"
            >
              Lorem Ipsum Dolor Sit Amet
            </p>
          </div>
        </li>
        <li>
          <div
            class="repo-details pointer space-between mt-4 border border-gray radius-3 p-4 bg-white "
            data-testid="repo-details"
          >
            <div
              class="info"
            >
              <div
                class="text-large"
              >
                <b>
                  Name
                </b>
                <i
                  class="material-icons "
                >
                  lock
                </i>
              </div>
              <div>
                <a
                  href="https://github.com/full/name"
                  target="_blank"
                >
                  full/name 
                  <i
                    class="material-icons "
                  >
                    open_in_new
                  </i>
                </a>
              </div>
            </div>
            <div
              class="star-count flex items-center text-large"
            >
              <i
                class="material-icons mr-2"
              >
                star
              </i>
              1
            </div>
            <div
              class="issue-count flex items-center text-large"
            >
              <i
                class="material-icons mr-2"
              >
                error_outline
              </i>
              3
            </div>
            <p
              class="mt-4 description"
            >
              Lorem Ipsum Dolor Sit Amet
            </p>
          </div>
        </li>
        <li>
          <div
            class="repo-details pointer space-between mt-4 border border-gray radius-3 p-4 bg-white "
            data-testid="repo-details"
          >
            <div
              class="info"
            >
              <div
                class="text-large"
              >
                <b>
                  Name
                </b>
              </div>
              <div>
                <a
                  href="https://github.com/full/name"
                  target="_blank"
                >
                  full/name 
                  <i
                    class="material-icons "
                  >
                    open_in_new
                  </i>
                </a>
              </div>
            </div>
            <div
              class="star-count flex items-center text-large"
            >
              <i
                class="material-icons mr-2"
              >
                star
              </i>
              1
            </div>
            <div
              class="issue-count flex items-center text-large"
            >
              <i
                class="material-icons mr-2"
              >
                error_outline
              </i>
              3
            </div>
            <p
              class="mt-4 description"
            >
              Lorem Ipsum Dolor Sit Amet
            </p>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
